src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -216,6 +218,8 @@ public function setup(): void
         } else {
             $this->driverConnection->getConfiguration()->setFilterSchemaAssetsExpression($assetFilter);
         }
+
+        $this->autoSetup = false;
```

Comment:
this could possibly be moved to the first line of the method incase `setup` fails?

Reply:
Hmm, I like it better here. It will then continue to retry... and probably fail... but it's a consistent behavior. Here at the bottom, setting it to false means "we have successfully set ourselves up".